### PR TITLE
Add geo location

### PR DESCRIPTION
**Pull request by prayag78**
### Add Geolocation to Fetch Coordinates 

1. Add geo location for fetch coordinate and then add them to api
2. Add default coordinates in case the user denies location access

Screenshot 2024-10-03 131015

IMG-20241003-WA0001

https://github.com/user-attachments/assets/8cb55355-c533-41a2-89e3-e6ad99d59156

### CODE REVIEW

**Review by mohitahlawat2001 — COMMENTED** [review]
not able to see food item , send photo with food items , after giving access to geo location

**Review by mohitahlawat2001 — CHANGES_REQUESTED** [review]
did to handle if api doesnt give data for any location outside of india 
what you should ,

**Comment by prayag78** [comment]
I just added the default Coordinates of india.

**Review by mohitahlawat2001 — CHANGES_REQUESTED** [review]
why you commented code , remove it

File: src/components/Body.js [hunk 1]
```diff
@@ -1,51 +1,179 @@
...
+//Added GeoLocation To get Users Location
+
+import React, { useEffect, useState } from "react";
 import { swiggyApi } from "../Constants";
 import RestaurantCard from "./RestaurantCard";
 import Shimmer from "./Shimmer";
 import { filterRestaurant } from "../utils/helper";
 
-
+const defaultLat = "21.1702401"; // Default latitude for India
```
Comment (mohitahlawat2001) [inline src/components/Body.js:111]: make these inside contants file

File: src/components/Body.js [hunk 2]
```diff
@@ -1,51 +1,179 @@
...
 const Body = () => {
-  const [allRestaurants, setAllRestaurants] = React.useState([]);
-  const [filteredRestaurants, setFilteredRestaurants] = React.useState([]);
-  const [search, setSearch] = React.useState("");
-  const [error, setError] = React.useState(null);
+  const [allRestaurants, setAllRestaurants] = useState([]);
+  const [filteredRestaurants, setFilteredRestaurants] = useState([]);
+  const [search, setSearch] = useState("");
+  const [error, setError] = useState(null);
+  const [location, setLocation] = useState({ lat: defaultLat, lon: defaultLon });
```
Comment (mohitahlawat2001) [inline src/components/Body.js:119]: use can use useMemo instead , more optimize

File: src/components/Body.js [hunk 3]
```diff
@@ -1,51 +1,179 @@
...
+      console.error("Geolocation not supported, using default location.");
+      getRestaurants(defaultLat, defaultLon); // Fetch with default location
+    }
+  };
+
+  // Fetch restaurants from API
+  const getRestaurants = async (lat, lon) => {
     try {
-      const response = await fetch(swiggyApi);
+      const apiUrl = `https://foodfire.onrender.com/api/restaurants?lat=${lat}&lng=${lon}&page_type=DESKTOP_WEB_LISTING`;
```
Comment (mohitahlawat2001) [inline src/components/Body.js:151]: take this url from constant file

**Review by mohitahlawat2001 — CHANGES_REQUESTED** [review]

**Review by mohitahlawat2001 — APPROVED** [review]

**Pull request merged**